### DISCUSSION TITLE
Starting point, goal point, and waypoints coordinate constraints

### DISCUSSION THREAD

**Discussion by Anas-Shrinah** [skhatiri/UAV-Testing-Competition#16] [Test Generation] [answered]
Is it safe to assume the following constraints:
1- The starting point x coordinate ranges from -40 to 30, and the y coordinate is less than 10.
2- The goal point x coordinate ranges from -40 to 30, and the y coordinate is either less than 10 or greater than 40.
3 - The waypoints x coordinate ranges from -40 to 30, and the y coordinate is either less than 10 or greater than 40.

Best regards,
Anas

**Comment by dgumenyuk** [comment]
@Anas-Shrinah thank you for the question. The waypoints for the drone's trajectory will be chosen outside the zone where obstacles are placed. The ranges you mentioned refer to the area for placing obstacles. The start, goal location and waypoints will be chosen outside of this area. To obtain the waypoints the drone is expected to complete, you can parse the mission .ulg file with [pyulog](https://github.com/PX4/pyulog).

**Reply by Anas-Shrinah** [reply]
Hi @dgumenyuk
Thanks for your answer. Actually, the ranges I mentioned are outside the obstacle-designated area -40 <  X  < 30  and y < 10 and y > 40.
Let me clarify my question:
I want to confirm that waypoints, goal points or start points will NOT be placed to the right or left of the obstacle designated area, i.e. the waypoints, goal point and start point are not placed with x > 30 or x < -40. Similarly, Their y coordinate is not allowed to be between 10 and 40.

Here is an example of waypoints located outside the designated area but placed to the left and right of the arena, which also has y less than 40 and greater than 10. Are such waypoints allowed?
![start_goal_points_example](https://github.com/user-attachments/assets/a9ef1947-ef50-4181-b89e-3ce5c2b37b3d)

**Reply by skhatiri** [reply, marked as answer]
The waypoints can be anywhere outside the obstacle area, including where you mentioned.
